app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :set_comment, only: %i[show edit update destroy]
+
+  # GET /comments or /comments.json
+  def index
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.all
+  end
+
+  # GET /comments/1 or /comments/1.json
+  def show
+    @post = Post.find(params[:post_id])
+    @comments = @post.comments.find(params[:id])
+  end
+
+  # GET /comments/new
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new
+  end
+
+  # GET /comments/1/edit
+  def edit
+    content_not_found if current_user.id != @comment.user_id
+
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.find(params[:id])
+  end
+
+  # POST /comments or /comments.json
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params)
+
+    @comment.user_id = current_user.id
+
+    respond_to do |format|
+      if @comment.save
+
+        format.turbo_stream
+        # format.html { redirect_to post_comments_path(@post), notice: "Comment was successfully created." }
+        format.html { redirect_to post_path(params[:post_id]), notice: 'Comment was successfully created.' }
+        format.json { render :show, status: :created, location: @comment }
+      else
+
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @comment.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /comments/1 or /comments/1.json
+  def update
+    respond_to do |format|
+      if @comment.update(comment_params)
+        format.html { redirect_to post_path(params[:post_id]), notice: 'Comment was successfully updated.' }
+        format.json { render :show, status: :ok, location: @comment }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @comment.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /comments/1 or /comments/1.json
+  def destroy
+    @comment.destroy
```

Comment:
> I can't see error handling is not done in cases such as destroying of records. 

@aliraza4638 The comments are not resolved. If you won't resolve comments and ask for re-review that will create frustration for the reviewer. The clients/devs won't like this at all.

Please ensure all comments I sent earlier are taken care of at all places.